keps/sig-multicluster/1645-multi-cluster-services-api/README.md

```diff
@@ -1020,7 +1023,9 @@ The conflict will be resolved by assigning precedence based on each
 A derived service will be accessible with the clusterset IP at the ports
 dictated by child services. If the external properties of service ports for a
 set of exported services don’t match, the clusterset service will expose the
-union of service ports declared on its constituent services.
+union of service ports declared on its constituent services and raise a `PortConflict`
+conflict condition. In that case, network traffic should be directed only to endpoints
+from constituent services that actually expose the targeted port.
```

Comment:
Can you clarify "targeted port", specifically in relation to the prior language that talks about "ports" (plural)? I assume by "targeted port" you mean a port that is in conflict, meaning one that is configured on one constituent service but not another. It sounds like you're saying the port should still be exposed but only from the constituent cluster that has it.

BTW, in Submariner, we only expose a port if it's configured for every constituent service.

Reply:
> Ah! But how does that works with the fact that the ServiceImport is exposing a union somehow?

The `ServiceImport` union is just for conformance - we don't use it. The real action is with the `EndpointSlices`.

Reply:
Hmm that sounds a bit hacky but in theory it sounds like the PortConflict would help signal this to your users then and as the network traffic part is only a recommendation you would still be conformant